engine/access/rpc/backend/backend_transactions.go

```diff
@@ -565,6 +565,65 @@ func (b *backendTransactions) GetTransactionResultByIndex(
 	}, nil
 }
 
+// GetSystemTransaction returns system transaction
+func (b *backendTransactions) GetSystemTransaction(ctx context.Context, _ flow.Identifier) (*flow.TransactionBody, error) {
+	systemTx, err := blueprints.SystemChunkTransaction(b.chainID.Chain())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get system chunk transaction: %v", err)
+	}
+
+	return systemTx, nil
+}
+
+// GetSystemTransactionResult returns system transaction result
+func (b *backendTransactions) GetSystemTransactionResult(ctx context.Context, blockID flow.Identifier, requiredEventEncodingVersion entities.EventEncodingVersion) (*access.TransactionResult, error) {
+	block, err := b.blocks.ByID(blockID)
+	if err != nil {
+		return nil, rpc.ConvertStorageError(err)
+	}
+
+	req := &execproto.GetTransactionsByBlockIDRequest{
+		BlockId: blockID[:],
+	}
+	execNodes, err := executionNodesForBlockID(ctx, blockID, b.executionReceipts, b.state, b.log)
+	if err != nil {
+		if IsInsufficientExecutionReceipts(err) {
+			return nil, status.Errorf(codes.NotFound, err.Error())
+		}
+		return nil, rpc.ConvertError(err, "failed to retrieve result from any execution node", codes.Internal)
+	}
+
+	resp, err := b.getTransactionResultsByBlockIDFromAnyExeNode(ctx, execNodes, req)
+	if err != nil {
+		return nil, rpc.ConvertError(err, "failed to retrieve result from execution node", codes.Internal)
+	}
+
+	systemTx, err := blueprints.SystemChunkTransaction(b.chainID.Chain())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get system chunk transaction: %v", err)
+	}
+	systemTxResult := resp.TransactionResults[len(resp.TransactionResults)-1]
```

Comment:
Can we add a check to make sure that we have retrieved tx result for system tx. Something along the lines:
```
systemTxID := systemTx.ID()
if systemTxResult.TransactionID != systemTxID {
	// bail out with internal error
}
```
`systemTxID` can be used in response later.